scripts/generate-docs.sh

```diff
@@ -33,31 +33,67 @@ currDir=$(pwd)
 baseDir=$(git rev-parse --show-toplevel)
 
 # Generation variables.
-mode="modules"
-tsconfig="tsconfig.json"
-urlPrefix="https://github.com/tensorflow/magenta-js/tree/master/${PKG_NAME}/src/"
+urlPrefix="https://github.com/${ORG_NAME}/magenta-js/tree/master/${PKG_NAME}/src/"
+keepAfter="/src/"
+scriptToFixTheToc=""
+
 if [ $PKG_NAME == "image" ]
 then
-  mode="file"
   urlPrefix="$urlPrefix/arbitrary_stylization/"
+  keepAfter="/arbitrary_stylization/"
 elif [ $PKG_NAME == "music" ]
 then
-  tsconfig="tsconfig.es5.json"
+  # The root index.ts file has a bunch of "export * from './foo';" lines.
+  # Parse those lines into a space separated list of names. It's ok that
+  # they're space separated, we'll split them in JS, this is all a horror anyway.
+  exports=`sed -n "s/export \* from '.\/\(.*\)';/\1/p" $currDir/src/index.ts`
+  scriptToFixTheToc="<script> \
+const toc = \"$exports\".split(' '); \
+const links = document.querySelectorAll('.tsd-kind-external-module'); \
+for (let i = 0; i < links.length; i++) { \
+  const name = links[i].textContent.trim().replace(/\"/g, ''); \
+  if (toc.indexOf(name) === -1) { \
+    links[i].parentNode.removeChild(links[i]); \
+  } \
+} \
+</script>"
 fi
 
 # Generate the docs.
 rm -rf $tmpDir
-npx typedoc --tsconfig $tsconfig --sourcefile-url-prefix $urlPrefix --out $tmpDir  --mode $mode --excludePrivate --exclude '**/*+(index|test|lib).ts' --excludeExternals src
+npx typedoc --options typedoc.json src --out $tmpDir
+
+###
+# Here begin the different hacks to fix the docs because typedoc has a lot of bugs.
+###
+
+# Typedoc generates documentation for _all_ files, not just the ones
+# actually exported in the library, so insert a script to fix the index.html.
+# see https://github.com/TypeStrong/typedoc/issues/639
+echo $scriptToFixTheToc >> $tmpDir/index.html
```

Comment:
The builtin TOC option requires that the names be exact, including the quotes since they are module names.

What you are after is this:

```
{
  "toc": [
    "\"coconet\"",
    "\"core\"",
    ....
  ],
  other options
}
```

Reply:
Oh brilliant, would you mind if I sent a PR to update this in the docs? This is super confusing

Reply:
That would be great!